dask_cuda/cli/dask_cuda_worker.py

```diff
@@ -121,8 +121,8 @@
 
     .. note::
         The asynchronous allocator requires CUDA Toolkit 11.2 or newer. It is also
-        incompatible with RMM pools and managed memory, and will be preferred over them
-        if both are enabled.""",
+        incompatible with RMM pools and managed memory, trying to enable both will
+        result in an exception.""",
```

Comment:
Since this is the CLI, is this really going to raise an exception or fail with an error instead? I'm not really sure what happens when an exception is raise in the `CUDAWorker` class when running the CLI tool.

Reply:
It looks like this fails with an error - I mostly just copied the phrasing used to describe the behavior of using NVLink + managed memory:

https://github.com/rapidsai/dask-cuda/blob/1526017749ae0a3137c40605bb992d426d9fee3b/dask_cuda/cli/dask_cuda_worker.py#L112-L114

I can change this here and in #561 if it would make more sense to say the CLI will fail outright in these cases.